framework/src/mediaplayer/mediaplayer.c

```diff
@@ -0,0 +1,530 @@
+{
+	return MEDIA_OK;
+}
+
+int media_setposition(media_t *m, int time_ms)
+{
+	return MEDIA_OK;
+}
+
+int media_getposition(media_t *m)
+{
+	return MEDIA_OK;
+}
+
+/***
+ * Functions run on threads
+ */
+
+static int read_wav_header(media_t *m)
+{
+	//struct pcm_config config;
+	struct wav_header_s header;
+	int ret;
+
+	ret = read(m->fd, &header, sizeof(header));
+	if (ret != sizeof(header)) {
+		return -MEDIA_ERROR_UNKNOWN_FILE;
+	}
+
+	// Todo: Add header information to media_t
+
+	return MEDIA_OK;
+}
+
+media_t *media_open(char *path, media_op_e op, media_type_e type)
+{
+	media_t *m = (media_t *)malloc(sizeof(media_t));
+	if (m == NULL) {
+		return NULL;
+	}
+
+	m->fd = -1;
+	m->op = op;
+	m->state = MEDIA_STATE_CREATED;
+	m->type = type;
+
+	if (type == MEDIA_TYPE_WAV || type == MEDIA_TYPE_PCM) {
+		if (op == MEDIA_OP_PLAYBACK) {
+			m->fd = open(path, O_RDONLY);
+			if (m->fd < 0) {
+				return NULL;
+			}
+
+			if (m->type == MEDIA_TYPE_WAV) {
+				read_wav_header(m);
+			}
+			m->state = MEDIA_STATE_PLAYING;
+
+		} else if (op == MEDIA_OP_RECORD) {
+			m->fd = open(path, O_RDWR | O_CREAT | O_TRUNC);
+			if (m->fd < 0) {
+				return NULL;
+			}
+			m->state = MEDIA_STATE_RECORDING;
+		}
+	} else if (type == MEDIA_TYPE_WAV_STREAM) {
+		// Todo: Supports network streaming
+		return NULL;
+	}
+
+	return m;
+}
+
+void media_close(media_t *m)
+{
+	if (m == NULL) {
+		return;
+	}
+	if (m->fd >= 0) {
+		close(m->fd);
+	}
+	free(m);
+	return;
+}
+
+
+int on_media_state_playing(struct pcm *pcmout, media_t *m, char *buffer, unsigned int buffer_size)
+{
+	int readed;
+	int remain;
+	int ret;
+	readed = read(m->fd, buffer, buffer_size);
+	if (readed) {
+		remain = pcm_bytes_to_frames(pcmout, readed);
+		while (remain > 0) {
+			ret = pcm_writei(pcmout, buffer + readed - pcm_frames_to_bytes(pcmout, remain), remain);
+			if (ret <= 0) {
+				return -MEDIA_ERROR_PCM_WRITE;
+			} else {
+				remain -= ret;
+			}
+		}
+	}
+
+	return readed;
+}
+
+int on_media_state_recording(struct pcm *pcmin, media_t *m, char *buffer, unsigned int buffer_size)
+{
+	int readed;
+	int remain;
+	int ret;
+
+	readed = pcm_readi(pcmin, buffer, pcm_bytes_to_frames(pcmin, buffer_size));
+	if (readed) {
+		remain = pcm_frames_to_bytes(pcmin, readed);
+		while (remain > 0) {
+			ret = write(m->fd, buffer + pcm_frames_to_bytes(pcmin, readed) - remain, remain);
+			if (ret < 0) {
+				return -MEDIA_ERROR_WRITE_TO_FILE;
+			} else {
+				remain -= ret;
+			}
+		}
+	}
+	return readed;
+}
+
+/***
+ * Thread routine
+ */
+int threadfunc_audio_playing(void *args)
+{
+	char *buffer;
+	unsigned int buffer_size;
+	struct pcm *pcmout;
+	struct pcm_config config;
+	dq_entry_t *node;
+	int ret;
+#ifndef CONFIG_AUDIO_MULTI_CARD
+	struct pcm *pcmin;
+	int playing = 0;
+	int recording = 0;
+	buffer = NULL;
+	buffer_size = 0;
+	pcmout = NULL;
+	pcmin = NULL;
+#endif
+	// Configuration
+	config.channels = 2;
+	config.rate = 16000;
+	config.period_size = CONFIG_AUDIO_BUFFER_NUMBYTES;
```

Comment:
I changed this configurations in my commits.
Also it had better get these informations from application....

Reply:
We are waiting to get clarifications on #717 . Since it changes the way buffer is allocated and there was an idea to remove pcm_set_config from public API list. (It has been declared as static function in Tizen 1.1 public release.) So after #717 is decided upon , then we can decide policy of calling pcm_set_config. 
I plan to take these values from application itself but we need to decide firstly about pcm_set_config.